### ISSUE TITLE
Ran into issues with ember-data 1.0.0-beta.19+

### ISSUE THREAD

**Issue by jpadilla** [lazybensch/ember-cli-filter-by-query#5]
Just ran into issues with ember-data 1.0.0-beta.19+ when upgrading emberaddons.com.  We use `ember-cli-filter-by-query`'s `computedFilterByQuery` passing `model.content`. We get an `DS.Model`  instead of a `InternalModel` in 1.0.0-beta.18.

Manage to put together two examples: 1.0.0-beta.19.1: http://emberjs.jsbin.com/behulememe/1/edit?js,output and 1.0.0-beta.18 http://emberjs.jsbin.com/fugohequke/1/edit?js,output

Note: I have a simple `computedFilterByQuery` method that represents the problem.

Removing `.content` and the `Ember.makeArray` seems to "fix" it.

``` js
var computedFilterByQuery = function(dependentKey, propertyKeys, queryKey, options) {
  propertyKeys = Ember.makeArray(propertyKeys);

  return Ember.computed(queryKey, '' + dependentKey + '.@each.{' + propertyKeys.join(',') + '}', function() {

    var array = this.get(dependentKey);
    var query = this.get(queryKey) || '';

    return filterByQuery(array, propertyKeys, query, options);

  });
};
```

**Comment by lazybensch** [comment]
thx for reporting @jpadilla ! i will have a look into that tomorrow and push a solution

**Comment by jpadilla** [comment]
@lazybensch I also ran into issue further down the road with `filterByQuery`. With the change I proposed up top I had to obviously do `return array.objectAt(item.id);` instead of `array[item.id]`.

**Comment by broerse** [comment]
I fixed it like this https://github.com/broerse/ember-cli-blog/commit/cbddc1ce1452bcd2244e912974e284f44b819e02 before I saw it was already reported. I think it is better if we don't have to wrap it in a function.

**Comment by broerse** [comment]
For an other project with ember I only changed the `return array.objectAt(item.id);` and everything worked.

@jpadilla it seems you don't have to sort again if you do this:

``` javascript
sifter = new Sifter(input);
sifter.getSortFunction = function() {
  return null;
};
result = sifter.search(query, options);
```

**Comment by lazybensch** [comment]
@jpadilla thx again! i published the fix under `ember-cli-filter-by-query@1.0.3`
sorry for the delay, i was quite occupied with day job stuff :(
